src/scmrepo/git/__init__.py

```diff
@@ -266,17 +266,19 @@ def no_commits(self):
     # https://github.com/iterative/dvc/issues/5641
     # https://github.com/iterative/dvc/issues/7458
     def _backend_func(self, name, *args, **kwargs):
-        for key, backend in self.backends.items():
+        backends: Iterable[str] = kwargs.pop("backends", self.backends)
+        for key in backends:
             if self._last_backend is not None and key != self._last_backend:
                 self.backends[self._last_backend].close()
                 self._last_backend = None
             try:
+                backend = self.backends[key]
                 func = getattr(backend, name)
                 result = func(*args, **kwargs)
                 self._last_backend = key
                 self.backends.move_to_end(key, last=False)
                 return result
-            except NotImplementedError:
+            except (NotImplementedError, KeyError):
```

Comment:
What `KeyError` is being hit here? If someone tries to call this function with `backends=[...]` that contains an invalid backend we should really be raising that exception and not catching/ignoring it here

Reply:
It's related to the tests when we initialize the `backends` with `selected` argument, I had put the logic into `__getitem__` to raise different exceptions for the nonexistent backend names and the not initialized backend names.